### DISCUSSION TITLE
Issue labelling

### DISCUSSION THREAD

**Discussion by carlos-granados** [Behat/Behat#1496] [General]
I would like to start working on triaging the pending issues of the repository, closing those which are not valid any longer or have already been solved and labelling those which are still valid. Currently we have the following tags available in the repository:

- `Bug`
- `Dependencies`
- `Documentation`
- `Easy Pick`
- `Feature`
- `Help Wanted`
- `Maintenance`
- `Next Major`
- `Next Minor`
- `PHP 8`

Some of them like Bug or Feature seem pretty valid but I would question some of the others:
- What does "Help Wanted" exactly mean? Help wanted by whom? Does not make much sense to me
- I understand the need for "Next major". Big new features or BC breaking features should go in a new major, but can't really think why "Next minor" would be useful
- Do we really need a specific "PHP 8" label?

Apart from this, there is other information that I think we could try to capture.
- Regarding bugs, I like to know when a bug has been confirmed or has been reproduced by one of the maintainers. So I'ld like to have a "confirmed" or "reproduced" label
- What about urgency or criticality? Do you think we should have a way to mark this?
- It seems that in the past we also had an `idea` label. Do you think it might be useful to have such label?

Please let me know what you all think

**Comment by acoulton** [comment]
Thanks for this @carlos-granados 

* I think the idea of Help Wanted? is things we might want the community to work on? But realistically at the moment I think we want people to work on anything / everything so it's not a useful label.
* "Easy Pick" I assume is similar to "Help Wanted" as a label for things that might be good for a new contributor to try? It could be useful although I'm not sure I know the codebase well enough to be confident giving things that label yet.
* I agree Next minor is pointless. Next Major is only really needed to identify things that are currently blocked (so we can filter them out).
* Can't see any point of a "PHP 8" label.
* I agree, "confirmed" / "reproduced" is good
* Urgency might be useful for prioritisation, but maybe only with a single "Critical" label? I feel like realistically with the resource we have "Critical" will be tackled and anything else will wait for someone to feel like taking it on, so it might not be worth thinking too hard about categorising bugs that aren't Critical?
* Some sort of label for "idea" / "blue sky thinking" is probably a good idea, again to help us focus on things that are realistic to tackle in the short term?

**Reply by ttomdewit** [reply]
I don’t have anything to add to this. Very good reply. I wouldn’t want to lose “easy pick” but nothing really is at the moment. Might want to add it back in the future.

**Comment by carlos-granados** [comment]
@acoulton what do you think about an `In progress` label? That would let us label those issues for which work is already being done or for which there is already a PR in place so that when looking for new issues to tackle you may pass over those

**Reply by acoulton** [reply]
Good plan

**Reply by ttomdewit** [reply]
Seconded

**Comment by carlos-granados** [comment]
@acoulton @ttomdewit I have done a first pass through the issue list, labelling all issues which were not tagged and closing some of the issues which did not seem valid. Please let me know if you believe any of the issues should be tagged differently

My next priority would be to start going through the issues labelled as `bug` to confirm if they can be reproduced and to see it they are still valid

**Reply by ttomdewit** [reply]
Let me know where I can help and best put my effort with confirming bugs.

**Reply by acoulton** [reply]
Brilliant thanks @carlos-granados I'll have a look

**Comment by carlos-granados** [comment]
I have now gone through all the issues labelled as bugs and which were not in progress and either confirmed them or closed them if they could not be reproduced any longer
